### PR TITLE
Arch packaging updates

### PR DESCRIPTION
**Pull request by cgutman**
## Description
This updates our packaging for Arch-based distros to:
- Build kmsgrab by default for proper Wayland capture support on GNOME, KDE, and other non-wlroots-based compositors
- Add the post-install commands to make kmsgrab work out of the box
- Add optional dependencies for AMD and Intel VAAPI drivers for hardware encoding support

However, this requires removing the the standalone PKGBUILD artifact from our releases due to the new dependency on the install script. However, this should be fine. Users that want to build from source should use the AUR, while users that want the binary package directly from us will still be able to get that.

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: docker/archlinux.dockerfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ ENV COMMIT=${COMMIT}
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 # install dependencies
-# cuda, libcap, and libdrm are optional dependencies for PKGBUILD
+# cuda is an optional build-time dependency for PKGBUILD
 RUN <<_DEPS
```
Comment (ReenigneArcher) [inline docker/archlinux.dockerfile:38]: Can you add the new optional deps. I don't think it installs them automatically.

File: packaging/linux/Arch/sunshine.install [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  setcap cap_sys_admin+p $(readlink -f $(which sunshine))
+}
+
+post_install() {
+  do_setcap
+}
+
+post_upgrade() {
+  do_setcap
+}
```
Comment (ReenigneArcher) [inline packaging/linux/Arch/sunshine.install:11]:
```suggestion
}

```

**Review by ReenigneArcher — COMMENTED** [review]
Docs will also need to be updated in regards to installing the PKGBUILD

File: docker/archlinux.dockerfile (hunk 1 above)
Comment (cgutman) [reply docker/archlinux.dockerfile:38]: They aren't needed for the build, just running the package.

Comment (ReenigneArcher) [reply docker/archlinux.dockerfile:38]: Got it.

**Review by ReenigneArcher — APPROVED** [review]

**Comment by cgutman** [comment]
@KuleRucket if you're available for review too

**Comment by KuleRucket** [comment]
> @KuleRucket if you're available for review too

Looks good to me

**Comment by marlop352** [comment]
Couldn't an instruction to also download the install script be added instead of removing the instructions on how to use the PKGBUILD to build the package?

**Pull request merged**